core/src/server/helpers.rs

```diff
@@ -107,109 +107,81 @@ impl MethodSink {
 		self.tx.is_closed()
 	}
 
-	/// Send a JSON-RPC error to the client
-	pub fn send_error(&self, id: Id, error: ErrorObject) -> bool {
-		let json = match serde_json::to_string(&ErrorResponse::borrowed(error, id)) {
-			Ok(json) => json,
-			Err(err) => {
-				tracing::error!("Error serializing response: {:?}", err);
-
-				return false;
-			}
-		};
-
-		tx_log_from_str(&json, self.max_log_length);
-
-		if let Err(err) = self.send_raw(json) {
-			tracing::warn!("Error sending response {:?}", err);
-			false
-		} else {
-			true
-		}
+	/// Same as [`tokio::sync::mpsc::Sender::closed`].
+	///
+	/// # Cancel safety
+	/// This method is cancel safe. Once the channel is closed,
+	/// it stays closed forever and all future calls to closed will return immediately.
+	pub async fn closed(&self) {
+		self.tx.closed().await
 	}
 
-	/// Helper for sending the general purpose `Error` as a JSON-RPC errors to the client.
-	pub fn send_call_error(&self, id: Id, err: Error) -> bool {
-		self.send_error(id, err.into())
+	/// Get the max response size.
+	pub const fn max_response_size(&self) -> u32 {
```

Comment:
dq: Would it be useful for users to have a `max_log_length`?

Reply:
The `MethodSink` is not really exposed to users only the `SubscriptionSink`.

Thus, I guess not.